cmd/controller-manager/app/serve.go

```diff
@@ -33,11 +32,23 @@ import (
 	"k8s.io/kubernetes/pkg/util/configz"
 )
 
-type serveFunc func(handler http.Handler, shutdownTimeout time.Duration, stopCh <-chan struct{}) error
+// DefaultBuildHandlerChain takes in handler and CompletedConfig and returns a handler.
+func DefaultBuildHandlerChain(apiHandler http.Handler, c *CompletedConfig) http.Handler {
+	requestContextMapper := apirequest.NewRequestContextMapper()
+	requestInfoResolver := &apirequest.RequestInfoFactory{}
+	failedHandler := genericapifilters.Unauthorized(requestContextMapper, legacyscheme.Codecs, false)
+
+	handler := genericapifilters.WithAuthorization(apiHandler, requestContextMapper, c.Authorization.Authorizer, legacyscheme.Codecs)
+	handler = genericapifilters.WithAuthentication(handler, requestContextMapper, c.Authentication.Authenticator, failedHandler)
+	handler = genericapifilters.WithRequestInfo(handler, requestInfoResolver, requestContextMapper)
+	handler = apirequest.WithRequestContext(handler, requestContextMapper)
+	handler = genericfilters.WithPanicRecovery(handler)
+
+	return handler
+}
 
-// Serve creates a base handler chain for a controller manager. It runs the
-// the chain with the given serveFunc.
-func Serve(c *CompletedConfig, serveFunc serveFunc, stopCh <-chan struct{}) error {
+// NewDebugHandler takes in CompletedConfig and returns a handler.
+func NewDebugHandler(c *CompletedConfig) http.Handler {
```

Comment:
healthz is not really debug. Can we call it NewBaseHandler or something like that?

Reply:
ok to me